src/Platforms/OraclePlatform.php

```diff
@@ -596,9 +596,9 @@ private function normalizeIdentifier($name)
      */
     private function addSuffix(string $identifier, string $suffix): string
     {
-        $maxPossibleLengthWithoutSuffix = $this->getMaxIdentifierLength() - strlen($suffix);
-        if (strlen($identifier) > $maxPossibleLengthWithoutSuffix) {
-            $identifier = substr($identifier, 0, $maxPossibleLengthWithoutSuffix);
+        $max = $this->getMaxIdentifierLength() - strlen($suffix);
+        if (strlen($identifier) > $max) {
+            $identifier = substr($identifier, 0, $max - 4) . '_' . substr(hash('sha256', $identifier), -3);
```

Comment:
Please extract to `private const HASH_LENGTH = 4`,

Reply:
Done but used `HASH_LENGTH = 3`. Otherwise, the code would have looked like:
```
$identifier = substr($identifier, 0, $max - self::HASH_LENGTH) . '_' . substr(hash('sha256', $identifier), -self::HASH_LENGTH + 1);
```
Nobody would understand why it's `-self::HASH_LENGTH + 1` at first.

Reply:
Please see above:
> Given that each byte in the hash is represented by two numbers, the choice of 3 looks unexpected. Should we use say four instead?

And make it `4` without changing the logic of building the identifier.

Reply:
> > Given that each byte in the hash is represented by two numbers, the choice of 3 looks unexpected. Should we use say four instead?
> 
> And make it `4` without changing the logic of building the identifier.

It doesn't matter that bytes in hashes are represented by two characters because we don't need any bytes or align to any byte order. We just need a few characters to make the name unique and I would have used one or two chars to avoid changing existing names if it would be possible but that's not guarantied to be unique enough. The more characters we use, the more likely it is that an existing sequence name will change. Please focus on what we need and avoid BC breaks, not on personal preference.

What does "without changing the logic of building the identifier" mean?

Reply:
@aimeos your patches take an abnormally large amount of involvement on my side compared to their scope. As a first-time contributor, if instead of cooperating and learning about the process, you want to tell me what I should focus on, please close your pull requests and maintain these patches yourself.

> What does "without changing the logic of building the identifier" mean?

Please change `3` to `4`.